lib/period.js

```diff
@@ -1,66 +1,136 @@
-/** "Period" object to transform between diverse Period formats */
-function getTerm (period) {
-  if (period <= 2) {
-    return 'HT'
-  } else {
-    return 'VT'
+module.exports = class Period {
+  /**
+   * @param {object} obj
+   * @param {number} obj.year Year
+   * @param {string} obj.term "HT" or "VT"
+   * @param {number} obj.period
+   */
+  constructor ({ year, term, period }) {
+    if (period > 5) {
+      throw new Error(
+        `Wrong format. Period [P${period}] does not exist. Should be from 0 to 5`
+      )
+    }
+
+    if (Period.getTerm(period) !== term) {
+      throw new Error(
+        `Wrong format. [${term}-P${period}] does not exist. Valids are HT-P0, HT-P1, HT-P2, VT-P3, VT-P4, VT-P5`
+      )
+    }
+
+    this.year = year
+    this.term = term
+    this.period = period
   }
-}
 
-function prevPeriod ({ year, period }) {
-  const y2 = period - 1 === 2 ? year - 1 : year
-  const p2 = period - 1 < 0 ? 5 : period - 1
+  /** Alternative constructor from a String instead of passing "year", "term", "period" */
+  static fromString (str) {
+    const matching = str.match(/^(\d+)-(VT|HT)-P(\d)$/)
```

Comment:
I believe it would be better to use `\d{4}` instead of \d+ since otherwise 20-VT-P3 will pass the control, but I guess it will lead to other problems down the line.

Reply:
Well, you never know if we will use LMS Antagna in year 10000 ;)

Reply:
Are you fine with me changing this @exacs ?

Reply:
Sure!